### PR TITLE
Upgrade to dotnet8, and new plugin to list project dependencies

### PR DESCRIPTION
**Pull request by ErikApption**
the removal of useless packages doesn't track versions and can remove a higher version dependency simply because it is a transitive one. This small fix prevents this issue.

### CODE REVIEW

**Comment by ErikApption** [comment]
@microsoft-github-policy-service agree

**Review by Anduin2017 — DISMISSED** [review]

File: src/NugetNinja.UselessPackageReferencePlugin/Services/UselessPackageReferenceDetector.cs [hunk 1]
```diff
@@ -72,10 +72,58 @@ private async IAsyncEnumerable<UselessPackageReference> AnalyzeProject(Project c
...
         }
     }
+
+    private bool GreaterVersion(NugetVersion version1, NugetVersion version2)
+    {
+        return version1 > version2;
+    }
+
+    public static int CompareNuGetVersions(string version1, string version2)
+    {
```
Comment (Anduin2017) [inline src/NugetNinja.UselessPackageReferencePlugin/Services/UselessPackageReferenceDetector.cs]: What's this function for?

File: src/NugetNinja/Properties/launchSettings.json [hunk 2]
```diff
@@ -31,6 +31,15 @@
         "Run missing": {
             "commandName": "Project",
             "commandLineArgs": "fill-properties -p \"..\\..\\..\\..\\..\""
+        },
+        "Dryrun Clean Pkg DP": {
+            "commandName": "Project",
+            "commandLineArgs": "clean-pkg -p \"E:\\\\datahub-portal\" -v -d"
```
Comment (Anduin2017) [inline src/NugetNinja/Properties/launchSettings.json]: Seems this path may not exist on other developers' machines. Mind if removing?

**Comment by Anduin2017** [comment]
Don't forget here: https://github.com/microsoft/NugetNinja/blob/main/.github/workflows/build.yml

**Comment by ErikApption** [comment]
Bigger update... I moved solution to net 8 and since dotnet list package is broken with shproj, I decided to re-use this code to generate an excel file with all the packages.

**Comment by Anduin2017** [comment]
Maybe also updating this file is needed:

https://github.com/microsoft/NugetNinja/blob/main/.github/workflows/build.yml

**Comment by ErikApption** [comment]
> Maybe also updating this file is needed:
> 
> https://github.com/microsoft/NugetNinja/blob/main/.github/workflows/build.yml

done - can you rerun the workflows?

**Review by Anduin2017 — APPROVED** [review]

**Comment by Anduin2017** [comment]
Thank you for the contribution!!!

**Pull request merged**